### PR TITLE
[UPSTREAM] RHODS 3159 - PVC and cluster settings page updates

### PR DESCRIPTION
**Pull request by dlabaj**
- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] For commits that came from upstream, `[UPSTREAM]` has been prepended to the commit message
- [x] JIRA link(s): https://issues.redhat.com/browse/RHODS-3159 
- [x] The Jira story is acked
- [x] An entry has been added to the latest build document in [Build Announcements Folder](https://drive.google.com/drive/folders/1sgkK1WZgGo9CXsLizNe0GbAzVKuSKrZL).
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious)
- [x] The developer has manually tested the changes and verified that the changes work.

This PR adds the ability for a RHODS admin to change notebook image size.

Testing instructions:

RHODS admin testing:

Login to the RHODS dashboard as a RHODS admin.
Verify that Settings in the left hand navigation is present.
Verify that Cluster Settings is a menu item under Settings.
Navigate to Cluster Settings.
Verify that the PVC Size is set to the default size or 20 GiB.
Enter a value in the number input between 1 - 16384 GiB.  Hit Enter to save the image size. (NOTE: This change will cause juypterhub to restart.  It will take 30 seconds before juypterhub will be available.)
After 30 seconds go to Enabled applications launch JuypterHub.
Set up and start your notebook.
In the openshift web console verify the size note book image size is the size of the image value you set under cluster settings
Screen Shot 2022-03-10 at 8 08 02 PM
.

### CODE REVIEW

File: backend/src/routes/api/cluster-settings/clusterSettingsUtils.ts [hunk 1]
```diff
@@ -1,17 +1,21 @@
 import { FastifyRequest } from 'fastify';
+import { scaleDeploymentConfig } from '../../../utils/deployment';
 import { KubeFastifyInstance, ClusterSettings } from '../../../types';
 
+const juypterhubCfg = 'jupyterhub-cfg';
+const setgmentKeyCfg = 'rhods-segment-key-config';
```
Comment (anishasthana) [inline backend/src/routes/api/cluster-settings/clusterSettingsUtils.ts:6]: `setgmentKeyCfg` is the wrong spelling :sweat_smile: 
Should be `segmentKeyCfg`

File: .env [hunk 2]
```diff
@@ -14,4 +14,4 @@ SOURCE_REPOSITORY_URL=git@github.com:red-hat-data-services/odh-dashboard.git
 SOURCE_REPOSITORY_REF=master
 DOC_LINK ='https://access.redhat.com/documentation/en-us/red_hat_openshift_data_science'
 SUPPORT_LINK ='https://access.redhat.com/support/cases/#/case/new/open-case?caseCreate=true'
-ENABLED_APPS_CM = 'rhods-enabled-applications-config'
+ENABLED_APPS_CM = 'rhods-enabled-applications-config'
```
Comment (anishasthana) [inline .env:17]: Missing new line

**Review by anishasthana — CHANGES_REQUESTED** [review]

File: backend/src/routes/api/cluster-settings/clusterSettingsUtils.ts [hunk 3]
```diff
@@ -1,17 +1,21 @@
...
   fastify: KubeFastifyInstance,
   request: FastifyRequest,
-): Promise<ClusterSettings | string> => {
+): Promise<{ success: boolean; error: string }> => {
   const coreV1Api = fastify.kube.coreV1Api;
   const namespace = fastify.kube.namespace;
   const query = request.query as { [key: string]: string };
   try {
-    if (query.userTrackingEnabled) {
+    if (query.userTrackingEnabled && query.pvcSize) {
```
Comment (DaoDaoNoCode) [inline backend/src/routes/api/cluster-settings/clusterSettingsUtils.ts:16]: we may not need query.pvcSize here

File: backend/src/routes/api/status/index.ts [hunk 4]
```diff
@@ -12,6 +12,7 @@ const status = async (
   fastify: KubeFastifyInstance,
   request: FastifyRequest,
 ): Promise<{ kube: KubeStatus }> => {
+  const adminGroup = process.env.ADMIN_GROUP;
```
Comment (DaoDaoNoCode) [inline backend/src/routes/api/status/index.ts:15]: remove `adminGroup` here.

File: frontend/src/pages/clusterSettings/ClusterSettings.scss [hunk 5]
```diff
@@ -1,7 +1,20 @@
...
+
+    .pf-c-input-group {
+      padding-top: var(--pf-global--spacer--md);
+      padding-bottom: var(--pf-global--spacer--md);
+    }
+
+    &__form {
+      margin: 0 var(--pf-global--spacer--lg);
+    }
   }
```
Comment (DaoDaoNoCode) [inline frontend/src/pages/clusterSettings/ClusterSettings.scss:20]: some problems with the indent and format of this file.

File: frontend/src/pages/clusterSettings/ClusterSettings.tsx [hunk 6]
```diff
@@ -49,51 +101,97 @@ const ClusterSettings: React.FC = () => {
...
-                  id="usage-data-checkbox"
-                  name="usageDataCheckbox"
-                />
-              </FormGroup>
-            </Form>
+                  >
+                    <Checkbox
+                      label="Allow collection of usage data"
+                      isChecked={clusterSettings.userTrackingEnabled}
+                      onChange={async () => {
```
Comment (DaoDaoNoCode) [inline frontend/src/pages/clusterSettings/ClusterSettings.tsx:183]: no need to use `async` here.

**Review by DaoDaoNoCode — COMMENTED** [review]
Left some comments, overall it LGTM

File: backend/src/routes/api/cluster-settings/clusterSettingsUtils.ts (hunk 1 above)
Comment (dlabaj) [reply backend/src/routes/api/cluster-settings/clusterSettingsUtils.ts:6]: done.

File: backend/src/routes/api/status/index.ts (hunk 4 above)
Comment (dlabaj) [reply backend/src/routes/api/status/index.ts:15]: done

File: backend/src/routes/api/cluster-settings/clusterSettingsUtils.ts (hunk 3 above)
Comment (dlabaj) [reply backend/src/routes/api/cluster-settings/clusterSettingsUtils.ts:16]: done

File: frontend/src/pages/clusterSettings/ClusterSettings.scss (hunk 5 above)
Comment (dlabaj) [reply frontend/src/pages/clusterSettings/ClusterSettings.scss:19]: Done.

File: .env (hunk 2 above)
Comment (dlabaj) [reply .env]: done.

**Review by anishasthana — APPROVED** [review]
/lgtm

**Comment by anishasthana** [comment]
(This is still waiting on final testing from the live build)

**Comment by crobby** [comment]
Live build for anyone interested:  quay.io/croberts/rhods-operator-live-catalog:1.8.0-0-jira-0000

**Review by crobby — APPROVED** [review]
/lgtm

**Comment by anishasthana** [comment]
Live build test on OSD looked good. /lgtm

**Pull request merged**